EpikCord/client/http_client.py

```diff
@@ -161,14 +164,30 @@ async def request(self, method: str, url: str, *args,  discord: bool = True, cha
         if url.endswith("/"):
             url = url[:-1]
         
-        bucket = self.buckets.get(url) or MockBucket()
+        bucket = self.buckets.get(f"{method}:{url}") or MockBucket()
 
         await self.global_event.wait()
         await bucket.wait()
 
         async with self.session.request(method, url, *args, **kwargs) as response:
```

Comment:
There is way too much logic going on this single async with. 
This already need refactoring...

Reply:
> If you're going for a rewrite you must define strong rules from the beginning to follow. It would be nice to set a linter and ensure mypy compliance for example. Before it becames too late

I was thinking of using PyRight instead, MyPy is confusing with its errors. It defaults to `object` when using a key that is missing in a `TypedDict`, PyRight does something similar iirc. I'll continue looking for a type checker.